### ISSUE TITLE
[API] Should encode() overwrite or update?

### ISSUE THREAD

**Issue by jakevdp** [ellisonbg/altair#101]
Consider this:
```python
layer = Layer(data)
layer.encode(x='Horsepower')
layer.encode(y='Miles_per_Gallon')
```

Currently, this result is identical to
```python
Layer(data).encode(y='Miles_per_Gallon')
```
this is because every time ``encode`` is called, it *overwrites* the current encoding.

I think it might be less confusing to change ``encode`` so that it instead *updates* the encoding, such that the above would be equivalent to 

```python
layer = Layer(data).encode(
   x='Horsepower',
   y='Miles_per_Gallon'
)
```
Thoughts?

**Comment by ellisonbg** [comment]
+1

**Comment by kanitw** [comment]
And If not too much more work, throw warning if users encode the same channel twice?

**Comment by jakevdp** [comment]
Throwing a warning will take a little bit more thought. Basic implementation is in #103

**Comment by ellisonbg** [comment]
I think if someone is iterating on a single Layer during exploration, it is very likely they will want to re-encode a single channel. Because of this, we shouldn't warn upon re-encoding a channel.

**Comment by jakevdp** [comment]
closed by #103
